### DISCUSSION TITLE
When is physical read amount smaller than physical written

### DISCUSSION THREAD

**Discussion by wangyuyue** [facebook/CacheLib#219] [General]
Dear developers,
I modified the cachebench code to also print  `NVM bytes read (physical)` by adding `ret.numNvmBytesRead = lookup("navy_device_bytes_read")` to my code.
Then I used it on `flat_kvcache_reg` workload. The result surprised me that NVM bytes written (physical) are 1.24 GB, while NVM bytes read (physical) are 1.20 GB. I thought the written bytes should always be fewer than the read bytes because the write in the nvm cache is a read-modify-write sequence, and lookup operations only contribute to the read. Now I guess my assumption is wrong, so what do you think is the reason that writes> reads?
Thanks very much.

**Comment by jaesoo-fb** [comment]
Hi @wangyuyue 

Yes, you are right in that the bytes read should always be greater than the bytes written in STEADY state where both BigHash and BlockCache needs to reclaim the space to make writes. What is the size of the NVM cache? Were those numbers captures after the NVM cache had been filled?

**Comment by pbhandar2** [comment]
Closing this discussion as it has been inactive. If this is still relevant, then please create a new discussion post and we will get back to you as soon as we can. Thank you!
